### PR TITLE
fix(api): refactor SupportService to use environment variable for API…

### PR DESCRIPTION
**Pull request by merrcury**
This fixes the SupportService created on PlainClient

### CODE REVIEW

**Review by jainpawan21 — APPROVED** [review]

File: libs/application-generic/src/custom-providers/index.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export const distributedLockService = {
 export const supportService = {
   provide: SupportService,
   useFactory: async () => {
-    const service = new SupportService(process.env.PLAIN_SUPPORT_KEY);
+    const service = new SupportService();
 
     return service;
   },
```
Comment (rifont) [inline libs/application-generic/src/custom-providers/index.ts]: We can remove this custom provider now, it's not doing anything.

Comment (merrcury) [reply libs/application-generic/src/custom-providers/index.ts]: Thats what I reckon.

File: libs/application-generic/src/services/support.service.ts [hunk 2]
```diff
@@ -5,12 +5,11 @@ const LOG_CONTEXT = 'SupportService';
...
-  constructor(private plainKey?: string | null) {
-    if (this.plainKey) {
-      this.plainClient = new PlainClient({
-        apiKey: this.plainKey,
-      });
-    }
+  private readonly plainKey: string;
+  constructor() {
+    this.plainKey = process.env.PLAIN_SUPPORT_KEY;
+    this.plainClient = new PlainClient({ apiKey: this.plainKey });
```
Comment (rifont) [inline libs/application-generic/src/services/support.service.ts]: How does the constructor behave when `PLAIN_SUPPORT_KEY` is _not_ present? Does this throw an error?

If it errors, the community edition will face issues. I would suggest only instantiating the client when `plainKey` is present in the environment. Then, conditionally access the upsert method like `this.plainClient?.upsertCustomer` in the upsert method

Comment (jainpawan21) [reply libs/application-generic/src/services/support.service.ts]: @rifont SupportService is being used in SupportModule and SupportModule is imported only when in enterprise environment

Comment (merrcury) [reply libs/application-generic/src/services/support.service.ts]: Made the Changes. One more thing with respect to invocation
For Community User, Its governed in main web only
```
export const IS_NOVU_PROD_STAGING = !IS_SELF_HOSTED && !API_ROOT.includes('localhost');
```

File: libs/application-generic/src/custom-providers/index.ts [hunk 3]
```diff
@@ -22,7 +21,7 @@ export const featureFlagsService = {
 export const getFeatureFlag = {
   provide: GetFeatureFlag,
   useFactory: async (
-    featureFlagsServiceItem: FeatureFlagsService,
+    featureFlagsServiceItem: FeatureFlagsService
```
Comment (rifont) [inline libs/application-generic/src/custom-providers/index.ts]:
```suggestion
    featureFlagsServiceItem: FeatureFlagsService,
```

File: libs/application-generic/src/custom-providers/index.ts [hunk 4]
```diff
@@ -79,20 +78,11 @@ export const distributedLockService = {
       cacheInMemoryProviderService.useFactory();
 
     const service = new DistributedLockService(
-      factoryCacheInMemoryProviderService,
+      factoryCacheInMemoryProviderService
```
Comment (rifont) [inline libs/application-generic/src/custom-providers/index.ts]: Removing formatting change to keep the diff clean (side note, it's strange that a Linting error isn't thrown here, it does in my local. We should fix that separately.
```suggestion
      factoryCacheInMemoryProviderService,
```

File: libs/application-generic/src/services/support.service.ts [hunk 5]
```diff
@@ -41,7 +44,7 @@ export class SupportService {
       Logger.error(
         { emailAddress, fullName, error: res.error },
         res.error.message,
-        LOG_CONTEXT,
+        LOG_CONTEXT
```
Comment (rifont) [inline libs/application-generic/src/services/support.service.ts]:
```suggestion
        LOG_CONTEXT,
```

File: libs/application-generic/src/services/support.service.ts [hunk 6]
```diff
@@ -67,7 +70,7 @@ export class SupportService {
       Logger.error(
         { plainCustomerId, threadText, error: res.error },
         res.error.message,
-        LOG_CONTEXT,
+        LOG_CONTEXT
```
Comment (rifont) [inline libs/application-generic/src/services/support.service.ts]:
```suggestion
        LOG_CONTEXT,
```

**Review by rifont — APPROVED** [review]
💯

File: libs/application-generic/src/services/support.service.ts (hunk 2 above)
Comment (rifont) [reply libs/application-generic/src/services/support.service.ts]: Thanks for clarifying, team. That makes sense.

**Pull request merged**